File: webapi/Models/CustomExercise.cs
namespace webapi.Models
{
    public class CustomExerciseData
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int ExerciseId { get; set; }
        public string ExerciseName { get; set; } = string.Empty;
        public string ExerciseType { get; set; } = string.Empty;
        public string? CompletionData { get; set; }
        public DateTime CompletedAt { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: reactapp/src/types/customExercise.ts
// Custom Exercise API Types
export interface ExerciseCompletionDto {
  userId: number;
  exerciseId: number;
  exerciseName: string;
  exerciseType: string;
  completionData?: Record<string, any>;
}

export interface ExerciseStatusDto {
  hasCompleted: boolean;
  completedAt?: string;
  completionData?: Record<string, any>;
}

export interface ExerciseCompletionResultDto {
  success: boolean;
  message: string;
}

export interface CustomExerciseData {
  id: number;
  userId: number;
  exerciseId: number;
  exerciseName: string;
  exerciseType: string;
  completedAt: string;
  completionData?: string;
}
